fitbenchmarking/fitbenchmarking
# Introduced a limit to the number of solvers in performance profile plots.

## Description of work

Fixes # 976

This PR limits the number of solvers shown in performance profiles.
When the number of solvers is higher than a certain number (15), the offline plots is not shown and the user is asked to rely on the Dash plot. The Dash plot will only allow to select 15 of the solvers that have been used, and they can be selected in the dropdown.


## Testing Instructions



1. Run fitbenchmarking with a large number of solvers (>15) to test.
2. After running fitbenchmarking, in the tables, the offline plot should not appear. There should be a red warning asking the user to rely on the offline plot.
3. After switching to offline plot, one should see that a maximum number of 15 solvers are shown. Under the dropdown, there should be a warning asking to unselect some solvers to select others, if needed. When the max number of solvers are shown in the plot, the others should appear as greyed out.
4.  Also the writing that generally appears under all performance profile plots should disappear when no offline plot is shown (and reappear when using the dash plot).

## Checklist
